src/ledger/database/lmdb.zig

```diff
@@ -0,0 +1,716 @@
+    /// Move cursor to prepare for #MDB_NEXT_MULTIPLE. Only for #MDB_DUPFIXED
+    get_multiple = 5,
+    /// Position at last key/data item
+    last = 6,
+    /// Position at last data item of current key. Only for #MDB_DUPSORT
+    last_dup = 7,
+    /// Position at next data item
+    next = 8,
+    /// Position at next data item of current key. Only for #MDB_DUPSORT
+    next_dup = 9,
+    /// Return up to a page of duplicate data items from next cursor position.
+    /// Move cursor to prepare for #MDB_NEXT_MULTIPLE. Only for #MDB_DUPFIXED
+    next_multiple = 10,
+    /// Position at first data item of next key
+    next_nodup = 11,
+    /// Position at previous data item
+    prev = 12,
+    /// Position at previous data item of current key. Only for #MDB_DUPSORT
+    prev_dup = 13,
+    /// Position at last data item of previous key
+    prev_nodup = 14,
+
+    /// Position at previous page and return up to a page of duplicate data items.
+    /// Only for #MDB_DUPFIXED
+    prev_multiple = 18,
+};
+
+fn result(int: isize) LmdbError!void {
+    return switch (int) {
+        0 => {},
+        -30799 => error.MDB_KEYEXIST,
+        -30798 => error.MDB_NOTFOUND,
+        -30797 => error.MDB_PAGE_NOTFOUND,
+        -30796 => error.MDB_CORRUPTED,
+        -30795 => error.MDB_PANIC,
+        -30794 => error.MDB_VERSION_MISMATCH,
+        -30793 => error.MDB_INVALID,
+        -30792 => error.MDB_MAP_FULL,
+        -30791 => error.MDB_DBS_FULL,
+        -30790 => error.MDB_READERS_FULL,
+        -30789 => error.MDB_TLS_FULL,
+        -30788 => error.MDB_TXN_FULL,
+        -30787 => error.MDB_CURSOR_FULL,
+        -30786 => error.MDB_PAGE_FULL,
+        -30785 => error.MDB_MAP_RESIZED,
+        -30784 => error.MDB_INCOMPATIBLE,
+        -30783 => error.MDB_BAD_RSLOT,
+        -30782 => error.MDB_BAD_TXN,
+        -30781 => error.MDB_BAD_VALSIZE,
+        -30780 => error.MDB_BAD_DBI,
+        -30779 => error.MDB_PROBLEM,
+        1 => error.EPERM,
+        2 => error.ENOENT,
+        3 => error.ESRCH,
+        4 => error.EINTR,
+        5 => error.EIO,
+        6 => error.ENXIO,
+        7 => error.E2BIG,
+        8 => error.ENOEXEC,
+        9 => error.EBADF,
+        10 => error.ECHILD,
+        11 => error.EAGAIN,
+        12 => error.ENOMEM,
+        13 => error.EACCES,
+        14 => error.EFAULT,
+        15 => error.ENOTBLK,
+        16 => error.EBUSY,
+        17 => error.EEXIST,
+        18 => error.EXDEV,
+        19 => error.ENODEV,
+        20 => error.ENOTDIR,
+        21 => error.EISDIR,
+        22 => error.EINVAL,
+        23 => error.ENFILE,
+        24 => error.EMFILE,
+        25 => error.ENOTTY,
+        26 => error.ETXTBSY,
+        27 => error.EFBIG,
+        28 => error.ENOSPC,
+        29 => error.ESPIPE,
+        30 => error.EROFS,
+        31 => error.EMLINK,
+        32 => error.EPIPE,
+        33 => error.EDOM,
+        34 => error.ERANGE,
```

Comment:
These numbers are technically platform dependent (yeah... :facepalm:)

![image](https://github.com/user-attachments/assets/0c48d473-9d7e-4544-b117-d05b642f3e48)

Reply:
Ah, I missed this, this is very incorrect. You need to integrate with `std.posix.errno`.

Reply:
229156237aa44950615578a02e4285f7d153b473